lib/extensions/ar_virtual.rb

```diff
@@ -597,6 +600,18 @@ def remove_virtual_fields(associations)
 # Class extensions
 #
 
+# this patch is no longer necessary for 5.2
+require "active_record/attribute"
+module ActiveRecord
+  # This is a bug in rails 5.0 and 5.1, but it is made much worse by virtual attributes
+  class Attribute
+    def with_value_from_database(value)
+      # self.class.from_database(name, value, type)
+      initialized? ? self.class.from_database(name, value, type) : self
```

Comment:
Wow, why?  Please provide a two line comment explaining why we need this.  I'm not sure of the why here.

Reply:
@kbrock your comment about says this is no longer needed for 5.2, can you explain why this is needed for 5.0 and 5.1?  You mention a bug... can you provide summary and link here?

Reply:
Also, do we need to remove this with 5.2?  Do we need a warning or something if this code is run against 5.2?

Reply:
This all changed in 5.2 - so yes, we will need to remove it.

I didn't know the proper way to warn. snippet reference?

Reply:
If we want to clean this up later, we can warn.  If we just want to patch it for 5.0 and 5.1, you can just put a guard clause  `return if Rails::VERSION >= "5.2.0" or something like that...

Reply:
If you want to just do this in a followup, I'm fine with that